hovor/actions/local_dialogue_action_simulated.py

```diff
@@ -0,0 +1,139 @@
+            return None
+        return random.choice(examples[category.upper()])
+
+    def get_random_value_for_var(self, var_name):
+        cv = self.data_for_sim['context_variables'][var_name]
+        if cv['type'] == 'enum':
+            return random.choice(cv['config'])
+        elif cv['type'] == 'json' and cv['config']['extraction']['method'] == 'spacy':
+            return self.example_spacy_entity(cv['config']['extraction']['config_method'])
+        else:
+            raise TypeError(
+                "Tried to fill in a non-enum or json var. Only enums and jsons are currently supported. ")
+
+    def fill_dollar_vars(self, s):
+        if '$' not in s:
+            return s
+        # finds the word starting with $ without the $
+        var_names = re.findall(r'\$(\w+)', s)
+        # get random values for these
+        random_var_values = [
+            self.get_random_value_for_var(n) for n in var_names]
+        # sub them into the string
+        for i in range(len(var_names)):
+            # only replace the first element incase there are two of same type vars
+            s = s.replace('$'+var_names[i], random_var_values[i], 1)
+        return s
+
+    def generate_response_by_data(self):
+        """
+        A function that picks a random outcome group from self,
+        and uses the information in data.json to find an appropriate response.
+        It will see if the intent is named directly, otherwise it will look at
+        the required entities and select an intent that has those entities.
+
+        This works well and produces a reasonable conversation. It works for intents
+        which don't have entities too. 
+        """
+        current_action_name = self.name
+        possible_outcomes = self.data_for_sim['actions'][current_action_name]['effect']['outcomes']
+        outcome_intent_names = [outcome['intent']
+                                for outcome in possible_outcomes]
+        possible_intent_names = [
+            intent_name for intent_name in outcome_intent_names if self.data_for_sim['intents'][intent_name]['type'] != 'fallback']
+        random_outcome_intent_name = random.choice(possible_intent_names)
+        if isinstance(random_outcome_intent_name, str):
+            # get the intent name and find the intent in data.json
+            # randomly pick an utterance
+            # use regex to replace variable with random variable from enum values
+            # in data.json
+            # done
+            random_intent = self.data_for_sim['intents'][random_outcome_intent_name]
+            if len(random_intent['utterances']) != 0:
+                random_utterance = random.choice(random_intent['utterances'])
+                simulated_input = self.fill_dollar_vars(random_utterance)
+            else:
+                # must be fallback, there are no utterances for this intent
+                simulated_input = "I'm not sure."
+        elif isinstance(random_outcome_intent_name, dict):
```

Comment:
If this is used to handle "intent dicts", I actually ended up phasing them out recently as they ended up being problematic for other reasons. So you are probably good to delete this whole block :)

Reply:
Oh okay that makes sense! When I was implementing the last changes with you recent changes pulled in, I couldn't seem to see where the dict intents would occur but I just knew I had encountered them. I will remove this then